network/endpoint.go

```diff
@@ -53,21 +53,26 @@ type endpoint struct {
 	PODName                  string `json:",omitempty"`
 	PODNameSpace             string `json:",omitempty"`
 	InfraVnetAddressSpace    string `json:",omitempty"`
-	NetNs                    string `json:",omitempty"`
+	NetNs                    string `json:",omitempty"` // used in windows
 	// SecondaryInterfaces is a map of interface name to InterfaceInfo
 	SecondaryInterfaces map[string]*InterfaceInfo
+	// Store nic type since we no longer populate SecondaryInterfaces
+	NICType cns.NICType
+	// Store the key used in stateless
+	NICName string
 }
 
 // EndpointInfo contains read-only information about an endpoint.
 type EndpointInfo struct {
-	Id                       string
+	EndpointID               string
 	ContainerID              string
 	NetNsPath                string
 	IfName                   string
 	SandboxKey               string
 	IfIndex                  int
 	MacAddress               net.HardwareAddr
-	DNS                      DNSInfo
+	EndpointDNS              DNSInfo
+	NetworkDNS               DNSInfo
```

Comment:
for dns, we can skip network dns and just have endpoint dns.. its fine if we dont configure dns in network creation.. it should still work since we set it up during endpoint creation.. @behzad-mir already validated this